src/Nethermind/Nethermind.Baseline/Tree/BaselineTree.cs

```diff
@@ -46,8 +64,9 @@ public BaselineTree(IDb db, IKeyValueStore metadataKeyValueStore, byte[] _dbPref
             TruncationLength = truncationLength;
             _db = db ?? throw new ArgumentNullException(nameof(db));
             _metadataKeyValueStore = metadataKeyValueStore ?? throw new ArgumentNullException(nameof(metadataKeyValueStore));
-
             this._dbPrefix = _dbPrefix;
+            Metadata = new BaselineTreeMetadata(metadataKeyValueStore, _dbPrefix);
```

Comment:
I think the idea was to decouple it and reverse dependencies. Not just putting it into another class.

Reply:
yes, let us leave it as a technical debt for the days after the presentation